src/awe/script/bytecode.cpp

```diff
@@ -309,4 +296,28 @@ void Bytecode::eq() {
 	spdlog::trace("eq");
 }
 
+inline std::vector<Variable> Bytecode::extractParameters(byte argsBytes) {
+	std::vector<Variable> arguments;
+	arguments.reserve(argsBytes);
+	signed short bytesRemaining = argsBytes;
+	while (bytesRemaining > 0) {
+		if (_stack.empty())
+			break;
+		
+		if (std::get_if<entt::entity>(&_stack.top())) {
+			bytesRemaining -= 2;
+		} else {
+			bytesRemaining--;
+		}
+
+		arguments.push_back(_stack.top());
+		_stack.pop();
+	}
+	arguments.shrink_to_fit();
+
+	if (bytesRemaining < 0)
+		throw CreateException("Parameter mismatch: expected to get {} values from stack but got {}", argsBytes, argsBytes - bytesRemaining);
```

Comment:
It might be beneficial to change the wording here, expressing that you want to get a certain amount of data blocks (4 bytes) here